packages/commonwealth/server/routes/activeThreads.ts

```diff
@@ -12,77 +12,70 @@ const activeThreads = async (
+      Number.isNaN(threads_per_topic) ||
+      threads_per_topic < MIN_THREADS_PER_TOPIC ||
+      threads_per_topic > MAX_THREADS_PER_TOPIC
+    ) {
+      threads_per_topic = 3;
+    }
 
-  const chain = req.chain;
+    const chain = req.chain;
 
-  const allThreads = [];
-  const communityWhere = { chain_id: chain.id };
-  const communityTopics = await models.Topic.findAll({
-    where: communityWhere,
-  });
+    const communityWhere = { chain_id: chain.id };
+    const communityTopics = await models.Topic.findAll({
+      where: communityWhere,
+    });
 
-  const threadInclude = [
-    { model: models.Address, as: 'Address' },
-    { model: models.Address, as: 'collaborators' },
-    { model: models.Topic, as: 'topic', required: true },
-  ];
+    const threadInclude = [
+      { model: models.Address, as: 'Address' },
+      { model: models.Address, as: 'collaborators' },
+      { model: models.Topic, as: 'topic', required: true },
+    ];
 
-  await Promise.all(
-    communityTopics.map(async (topic) => {
-      const recentTopicThreadsRaw = await models.Thread.findAll({
-        where: {
-          topic_id: topic.id,
-        },
-        include: threadInclude,
-        limit: threads_per_topic,
-        order: [
-          ['created_at', 'DESC'],
-          ['last_commented_on', 'DESC'],
-        ],
-      });
+    let allRecentTopicThreadsRaw = [];
+    allRecentTopicThreadsRaw = await Promise.all(
+      communityTopics.map(async (topic) => {
+        return await models.Thread.findAll({
+          where: {
+            topic_id: topic.id,
+          },
+          include: threadInclude,
+          limit: threads_per_topic,
+          order: [
+            ['created_at', 'DESC'],
+            ['last_commented_on', 'DESC'],
+          ],
+        });
+      })
+    );
 
-      const recentTopicThreads = recentTopicThreadsRaw.map((t) => {
-        return t.toJSON();
-      });
+    allRecentTopicThreadsRaw = allRecentTopicThreadsRaw.flat();
 
-      const threadsWithCommentsCount = await getThreadsWithCommentCount({
-        threads: recentTopicThreads,
-        models,
-        chainId: chain.id,
-      });
+    const allRecentTopicThreads = allRecentTopicThreadsRaw.map((t) => {
+      return t.toJSON();
+    });
 
-      // In absence of X threads with recent activity (comments),
-      // commentless threads are fetched and included as active
-      // if (!recentTopicThreads || recentTopicThreads.length < threads_per_topic) {
-      //   const commentlessTopicThreads = await models.OffchainThread.findAll({
-      //     where: {
-      //       topic_id: topic.id,
-      //       last_commented_on: {
-      //         [Op.is]: null,
-      //       }
-      //     },
-      //     include: threadInclude,
-      //     limit: threads_per_topic - (recentTopicThreads || []).length,
-      //     order: [['created_at', 'DESC']]
-      //   });
+    const allThreadsWithCommentsCount = await getThreadsWithCommentCount({
```

Comment:
This is where we are saving queries:

Queries saved is in nested function

`getThreadsWithCommentCount` - is called once for all topics now, rather than once per topic
Moved this up - to save queries we make to DB

Reply:
`communityTopics.map` - just group threads by topics in array of arrays